atsc/src/header.rs

```diff
@@ -14,25 +14,106 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-use bincode::{Decode, Encode};
+use log::{debug, trace};
+use std::panic;
 
-#[derive(Encode, Decode, Debug, Clone)]
+/*  The current file version.
+    On file read, compressors check the version and uncompress accordingly to that (of fail)
+*/
+const CURRENT_VERSION: u32 = 1;
+#[derive(Debug, Clone)]
 pub struct CompressorHeader {
     initial_segment: [u8; 4],
-    // We should go unsigned
-    frame_count: i16,
+    pub version: u32,
+    frame_count: u8,
+}
+
+fn verify_header_versions(version: u32) {
+    let current_version = CURRENT_VERSION;
+    trace!("Versions: c:{} h:{}", current_version, version);
+    match current_version.cmp(&version) {
+        std::cmp::Ordering::Less => panic!(
+            "Can't decompress! File is version ({}) is higher than compressor version ({})!",
+            version, current_version
+        ),
+        std::cmp::Ordering::Equal | std::cmp::Ordering::Greater => {
+            debug!("File version: {}", version)
+        }
+    }
 }
 
 impl CompressorHeader {
     pub fn new() -> Self {
         CompressorHeader {
             initial_segment: *b"BRRO",
```

Comment:
Personally I would just write the BRRO bytes directly in `to_bytes` and return an Err in `from_bytes` if the magic bytes are not BRRO.
But this is a very small nit either approach is fine.